controllers/configuration_controller.go

```diff
@@ -1454,81 +1454,91 @@ func (meta *TFConfigurationMeta) RenderConfiguration(configuration *v1beta2.Conf
-		return nil, errors.Wrap(err, errMsg)
-	}
-
 	needSecretKeys := []string{GitCredsKnownHosts, v1.SSHAuthPrivateKey}
-	for _, key := range needSecretKeys {
-		if _, ok := secret.Data[key]; !ok {
-			err := errors.Errorf("'%s' not in git credentials secret", key)
-			return nil, err
-		}
+	errMsg := "Failed to get git credentials secret"
+	keyErrMsg := "not in git credentials secret"
+	secret, _, err := GetSecretOrConfigMap(ctx, k8sClient, true, secretRef, needSecretKeys, errMsg, keyErrMsg)
+	if secret != nil {
+		return secret, nil
+	} else {
+		return nil, err
 	}
-
-	return secret, nil
 }
 
 // GetTerraformCredentialsSecret will get the secret containing the terraform credentials
 func GetTerraformCredentialsSecret(ctx context.Context, k8sClient client.Client, secretRef *v1.SecretReference) (*v1.Secret, error) {
-	secret := &v1.Secret{}
-	namespacedName := client.ObjectKey{Name: secretRef.Name, Namespace: secretRef.Namespace}
-	err := k8sClient.Get(ctx, namespacedName, secret)
-	if err != nil {
-		errMsg := "Failed to get terraform credentials secret"
-		klog.ErrorS(err, errMsg, "Name", secretRef.Name, "Namespace", secretRef.Namespace)
-		return nil, errors.Wrap(err, errMsg)
-	}
-
 	needSecretKeys := []string{TerraformCredentials}
-	for _, key := range needSecretKeys {
-		if _, ok := secret.Data[key]; !ok {
-			err := errors.Errorf("'%s' not in terraform credentials secret", key)
-			return nil, err
-		}
+	errMsg := "Failed to get terraform credentials secret"
+	keyErrMsg := "not in terraform credentials secret"
+	secret, _, err := GetSecretOrConfigMap(ctx, k8sClient, true, secretRef, needSecretKeys, errMsg, keyErrMsg)
+	if secret != nil {
+		return secret, nil
+	} else {
+		return nil, err
 	}
-
-	return secret, nil
-
 }
 
 // GetTerraformRegistryConfigMap will get the config map containing the terraform registry configuration
 func GetTerraformRegistryConfigMap(ctx context.Context, k8sClient client.Client, configMapRef *v1.SecretReference) (*v1.ConfigMap, error) {
-	configMap := &v1.ConfigMap{}
-	namespacedName := client.ObjectKey{Name: configMapRef.Name, Namespace: configMapRef.Namespace}
-	err := k8sClient.Get(ctx, namespacedName, configMap)
-	if err != nil {
-		errMsg := "Failed to get the terraform registry config configmap"
-		klog.ErrorS(err, errMsg, "Name", configMapRef.Name, "Namespace", configMapRef.Namespace)
-		return nil, errors.Wrap(err, errMsg)
-	}
 	neededKeys := []string{TerraformRegistryConfig}
-	for _, key := range neededKeys {
-		if _, ok := configMap.Data[key]; !ok {
-			err := errors.Errorf("'%s' not in terraform registry configuration configmap", key)
-			return nil, err
-		}
+	errMsg := "Failed to get the terraform registry config configmap"
+	keyErrMsg := "not in terraform registry configuration configmap"
+	_, configMap, err := GetSecretOrConfigMap(ctx, k8sClient, false, configMapRef, neededKeys, errMsg, keyErrMsg)
+	if configMap != nil {
+		return configMap, nil
+	} else {
+		return nil, err
```

Comment:
Well, we're used to check the err first. Sometimes you can assume that if configMap returned isn't nil, then no error happens. But it depends. My suggestion is to check it first.

Reply:
fixed